src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -340,36 +352,32 @@ def can_loop_be_parallelised(self, loop, loop_variable=None,
         if not var_accesses:
             var_accesses = VariablesAccessInfo()
             loop.reference_accesses(var_accesses)
-        if not variables_to_ignore:
-            variables_to_ignore = []
+        if not signatures_to_ignore:
+            signatures_to_ignore = []
 
         # Collect all variables used as loop variable:
         loop_vars = [loop.variable.name for loop in loop.walk(Loop)]
 
         result = True
         # Now check all variables used in the loop
         for signature in var_accesses.all_signatures:
+            # This string contains derived type information, e.g.
+            # "a%b"
+            var_string = str(signature)
             # Ignore all loop variables - they look like reductions because of
             # the write-read access in the loop:
-            var_name = str(signature)
-            if var_name in loop_vars:
+            if var_string in loop_vars:
                 continue
-            if var_name in variables_to_ignore:
+            if signature in signatures_to_ignore:
                 continue
 
+            # This returns the first component of the signature,
+            # i.e. in case of "a%b" it will only return "a"
+            var_name = signature.var_name
```

Comment:
I am not sure I see an advantage there. If I use the symbol instead of a string, I need to take care that different symbol instances with the same name result in the same hash code. That might already be the case (I am not sure if e.g. the loop variable symbol in the Loop class is the same instance that is stored in the symbol table or a different instance with the same name). Since we need a way of accessing the actual symbol name anyway, I think it's ok (for now).